xtask/src/build_ebpf.rs

```diff
@@ -0,0 +1,63 @@
+use std::{path::PathBuf, process::Command};
+
+use clap::Parser;
+
+#[derive(Debug, Copy, Clone)]
+pub enum Architecture {
+    BpfEl,
+    BpfEb,
+}
+
+impl std::str::FromStr for Architecture {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "bpfel-unknown-none" => Architecture::BpfEl,
+            "bpfeb-unknown-none" => Architecture::BpfEb,
+            _ => return Err("invalid target".to_owned()),
+        })
+    }
+}
+
+impl std::fmt::Display for Architecture {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(match self {
+            Architecture::BpfEl => "bpfel-unknown-none",
+            Architecture::BpfEb => "bpfeb-unknown-none",
+        })
+    }
+}
+
+#[derive(Debug, Parser)]
+pub struct Options {
+    /// Set the endianness of the BPF target
+    #[clap(default_value = "bpfel-unknown-none", long)]
+    pub target: Architecture,
+    /// Build the release target
+    #[clap(long)]
+    pub release: bool,
+}
+
+pub fn build_ebpf(opts: Options) -> Result<(), anyhow::Error> {
+    let dir = PathBuf::from("aurae-ebpf");
+    let target = format!("--target={}", opts.target);
+    let mut args =
+        vec!["build", "--verbose", target.as_str(), "-Z", "build-std=core"];
+    if opts.release {
+        args.push("--release")
+    }
+
+    // Command::new creates a child process which inherits all env variables. This means env
+    // vars set by the cargo xtask command are also inherited. RUSTUP_TOOLCHAIN is removed
+    // so the rust-toolchain.toml file in the -ebpf folder is honored.
+
+    let status = Command::new("cargo")
+        .current_dir(dir)
+        .env_remove("RUSTUP_TOOLCHAIN")
+        .args(&args)
+        .status()
+        .expect("failed to build bpf program");
```

Comment:
I'm getting rid of `xtask` altogether. I just started off with an `aya` template and moved that over to our codebase. I don't see any value keeping `xtask`, and as long as we're targeting `x86_64-*` we'll always have the little endian target (`bpfel-*`).

Reply:
Should we just build it in the root target folder, btw? Not sure what's idiomatic here since it is not a member of the workspace.

Reply:
What is the reasoning for not having it as part of the workspace?

Reply:
I'm not sure why but a lot of extra stuff was ending up in the binary when I included it in the workspace. I think it was like 10-20x times the size and `aya` wasn't able to perform the relocations on the `elf` file (the binary). I tried a few things but then I gave up as excluding it was easier. :) If you want to take a look and see what's going on, you could compile it as a workspace member and inspect the output file with `objdump -S <filename>`. All the names are mangled though so it can be a little hard on the eyes.

Reply:
If it can't be part of the workspace that is fine. It should have its own target directory.

Reply:
Cool, `xtask` is gone now and the `Makefile` updated. Note that we do have a dependency on `nightly` now just for this project only as well as a dependency on `bpf-linker`. Once this PR is in I'll update the docs in the `aurae.io` repo.

Reply:
This repo is also used for aurae.io. The other repo is archived.

Reply:
Yep found it already and included the changes in this PR.